tests/testthat/test-geom-rug.R

```diff
@@ -20,3 +20,26 @@ test_that("coord_flip flips the rugs", {
   expect_equal(length(b[[1]]$children[[1]]$y0), 1)
   expect_equal(length(b[[1]]$children[[1]]$y1), 1)
 })
+
+test_that("Rug length needs unit object", {
+  p <- ggplot(df, aes(x,y))
+  expect_is(p + geom_rug(length=grid::unit(0.01, "npc")), "ggplot")
```

Comment:
Sorry, one more comment here. This `expect_is()` seems no more needed. If `geom_rug()` raises an error with a `unit` object, the test case below will fail so that we can notice something is wrong.